packages/custom-form-field-scripts/src/scripts/dev.js

```diff
@@ -0,0 +1,45 @@
+import webpack from 'webpack'
+import WebpackDevServer from 'webpack-dev-server'
+import webpackConfig from '../config/webpack.config.js'
+import webpackDevServerConfig from '../config/webpackDevServer.config.js'
+import paths from '../paths.js'
+import chokidar from 'chokidar'
+import chalk from 'chalk'
+
+const runServer = async () => {
+    const compiler = webpack(webpackConfig)
+    const server = new WebpackDevServer(webpackDevServerConfig, compiler)
+    await server.start()
+    const watcher = chokidar.watch([paths.c3Config, paths.appPackageJson], {
+        persistent: true,
+    })
+
+    const killServer = async () => {
+        await server.stop()
+        process.exit(0)
+    }
+
+    watcher
+        .on('change', async (path) => {
+            console.log()
+            console.log(
+                chalk.bold.yellow(
+                    `The file at '${path}' was modified, rendering the build obsolete. Build killed! Please rerun the build.`
+                )
+            )
+            killServer()
```

Comment:
Recompiling is not adviced because the guy could have just added a package to devDependences or dependencies by editing the package.json and not using `npm i <package-name>`, so I thought this is a gracefully way of dealing with all the minute edge cases.